File: src/Emergence.Data.Shared/Stores/Photo.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Emergence.Data.Shared.Stores
{
    public class Photo : IAuditable
    {
        public int Id { get; set; }
        [StringLength(36)]
        public string Type { get; set; }
        public int? TypeId { get; set; }
        public int? LocationId { get; set; }
        [StringLength(200)]
        public string Filename { get; set; }
        [StringLength(500)]
        public string BlobPath { get; set; }
        [StringLength(100)]
        public string ContentType { get; set; }
        public int? Width { get; set; }
        public int? Height { get; set; }
        public DateTime? DateTaken { get; set; }
        [StringLength(36)]
        public string CreatedBy { get; set; }
        [StringLength(36)]
        public string ModifiedBy { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime? DateModified { get; set; }
        public Location Location { get; set; }
    }
}

File: src/Emergence.Web/ClientApp/src/app/shared/photo.ts
export interface Photo {
  photoId: number;
  type: number;
  typeId: number;
  filename: string;
  blobPath: string;
  contentType: string;
  width: number;
  height: number;
  dateTaken: Date;
  createdBy: string;
  dateCreated: Date;
  blobPathRoot: string;
  urlBroken: false;
  originalUri: string;
  largeUri: string;
  mediumUri: string;
  thumbnailUri: string;
}
